chess-AI/train.py

```diff
@@ -118,20 +142,13 @@ def create_dataset(games, move_approximator, val_approximator=None):
     return dataset
 
 
-def train_on_dataset(dataset, nnet, learning_rate, momentum, weight_decay, epochs, batch_size, device, save_path, num_saved_models, overwrite_save):
+def train_on_dataset(dataset, nnet, options):
     """Train with the specified dataset
 
     Attributes:
         dataset: the dataset to use for training
         nnet: the neural network
-        learning_rate: the learning rate for the optimizer
-        momentum: additional parameter for the optimizer
-        weight_decay: loss term encouraging smaller weights
-        epochs: number of epochs
-        batch_size: the batch size for SGD
-        device: the device being used to train (either CPU or GPU)
-        save_path: path for model checkpointing
-        num_saved_models: number of models to store
+        options: settings/hyperparameters for training
```

Comment:
Can you pls specify that the type is TrainOptions so people know where to look to see all the available hyperparameters